test/TestDataGenerator.cpp

```diff
@@ -0,0 +1,444 @@
+                          "not to overwrite it"));
+  } catch (...) {
+    FAIL() << "should have throw invalid_argument error.";
+  }
+}
+
+TEST(TestDataGenerator, TestXmlFileType) {
+  DataGenerator data = DataGenerator();
+  std::string type = data.getXmlFileType("test_model.xml");
+  std::ostringstream os2;
+  os2 << type;
+  EXPECT_EQ(os2.str(), "model");
+}
+
+TEST(TestDataGenerator, TestXmlFileType2) {
+  DataGenerator data = DataGenerator();
+  std::string type = data.getXmlFileType("test.xml");
+  std::ostringstream os2;
+  os2 << type;
+  EXPECT_EQ(os2.str(), "dataset");
+}
+
+TEST(TestDataGenerator, TestXmlFileType3) {
+  DataGenerator data = DataGenerator();
+  try {
+    std::string type = data.getXmlFileType("wrong_structure.xml");
+    FAIL() << "should have throw invalid_argument error.";
+  } catch (std::invalid_argument const &err) {
+    EXPECT_EQ(err.what(),
+              std::string("Cannot open xml file, please check path"));
+  } catch (...) {
+    FAIL() << "should have throw invalid_argument error.";
+  }
+}
+
+TEST(TestDataGenerator, TestLoadModel) {
+  DataGenerator data = DataGenerator();
+  std::tuple<float, Criteria, PerformanceTable> t =
+      data.loadModel("test_model.xml");
+  float lambda = std::get<0>(t);
+  Criteria crit = std::get<1>(t);
+  PerformanceTable pt = std::get<2>(t);
+  std::ostringstream os;
+  std::ostringstream os2;
+  std::ostringstream os3;
+  os << lambda;
+  os2 << crit;
+  os3 << pt;
+  EXPECT_EQ(os.str(), "0.9035");
+  EXPECT_EQ(os2.str(),
+            "Criteria(Criterion(id : crit0, direction : +, weight : 0.5), "
+            "Criterion(id : crit1, direction : +, weight : 0.5), )");
+  EXPECT_EQ(os3.str(),
+            "PerformanceTable(Performance(Perf( name : prof0, crit : crit0, "
+            "value : 0.93 ), Perf( name : prof0, crit : crit1, value : 0.93 ), "
+            "), Performance(Perf( name : prof1, crit : crit0, value : 0.93 ), "
+            "Perf( name : prof1, crit : crit1, value : 0.93 ), ), "
+            "Performance(Perf( name : prof2, crit : crit0, value : 0.93 ), "
+            "Perf( name : prof2, crit : crit1, value : 0.93 ), ), )");
+}
+
+TEST(TestDataGenerator, TestSaveModelWrongNumberCriteria) {
+  DataGenerator data = DataGenerator();
+  Criteria crit = Criteria(2, "a");
+  float lambda = 0.1;
+  PerformanceTable perf_table = PerformanceTable(3, crit, "test");
+  try {
+    data.saveModel("test_save_model.xml", lambda, crit, perf_table, 1);
+    FAIL() << "should have throw invalid_argument error.";
+  } catch (std::invalid_argument const &err) {
+    EXPECT_EQ(err.what(),
+              std::string(" Number of criteria and the number and the length "
+                          "of the performance of the fictive alternative (ie "
+                          "profile performance) does not match"));
+  } catch (...) {
+    FAIL() << "should have throw invalid_argument error.";
+  }
+}
+
+TEST(TestDataGenerator, TestSaveModel) {
```

Comment:
This is not testing anything, we should have a check somewhere. Also, let's not have random generated models populating the code base, if you want to test the save and check if the file was saved, do it under a tmp directory and delete the directory at the end of the test to make sure we are cleaning up after the mess

Reply:
YEs smart modified it !